server/src/api/commits.ts

```diff
@@ -16,10 +16,34 @@
 
 /**
  * @fileoverview Handles routing of /commits endpoints.
- * @author Karen Frilya Celine
  */
 
-import express from 'express';
-const router: express.Router = express.Router();
+import express, {Request, Response, NextFunction} from 'express';
 
-export const commitRouter: express.Router = router;
+import {commitService} from '../services';
+
+const commitRouter = express.Router();
+
+/**
+ * Handles the get request for retrieving all commits.
+ */
+commitRouter.get(
+  '/',
+  async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const queryParams = req.query;
+      if (Object.keys(queryParams).length === 0) {
+        res.sendStatus(400);
+        return;
+      }
+
+      const commits = await commitService.getAllCommits(queryParams);
+      res.status(200).json(commits);
+      // TODO: Add error handling with the appropriate response codes.
```

Comment:
Do you have an open issue for this TODO? Is it dealing with the various response codes like "400""200" in this case?

Reply:
@xlai20 Nope not those, but rather dealing with those that are not returning a specific error code as of now. Okay will add an issue!